### PR TITLE
CUMULUS-3135 - Update integration test scripts to fail on test timeout

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-3315](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3315)

## Changes
  - Update CI scripts to use shell logic/GNU timeout to bound test timeouts
    instead of NPM `parallel` package, as timeouts were not resulting in
    integration test failure

## PR Checklist

- [x] Update CHANGELOG
- [ ] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: example/scripts/run_test.sh [hunk 1]
```diff
@@ -2,22 +2,26 @@
...
+echo "$TIMESTAMP ../node_modules/.bin/jasmine $3 STARTED"
+timeout "$testTimeout" ../node_modules/.bin/jasmine --no-color "$3" > "$outputPath" 2>&1
 result=$?
 
 TIMESTAMP=$(date "+%Y-%m-%dT%H:%M:%S")
 if [ "$result" -eq "0" ]; then
-  echo "$TIMESTAMP ../node_modules/.bin/jasmine $2 PASSED"
+  echo "$TIMESTAMP ../node_modules/.bin/jasmine $3 PASSED"
+  mv "$outputPath" "$1/${specName}-passed.txt"
+elif { [ "$result" -gt "124" ] && [ "$result" -lt "128" ] ;} || [ "$result" -eq "137" ]; then
```
Comment (npauzenga) [inline example/scripts/run_test.sh:18]: What are we saying here?

**Review by npauzenga — CHANGES_REQUESTED** [review]
Looks good @Jkovarik. I'm marking "request changes" only because I'd like to understand why this doesn't work in the npm parallel package. Is that just a known issue we're working around or are we missing some configuration somewhere?

File: example/scripts/tests-parallel.sh [hunk 2]
```diff
@@ -10,11 +10,12 @@ DOT_PID="$!"
 
 TESTS=$(find spec/parallel -type f -name '*spec.js' -or -name '*Spec.js')
 testOutputDir=scripts/test_output
+testTimeout=1200
 
 rm -r -f $testOutputDir
 mkdir -p $testOutputDir
 
-echo "" | ../node_modules/.bin/parallel -j "${INTEGRATION_CONCURRENCY:=0}" --timeout 1200 sh scripts/run_test.sh  $testOutputDir ::: $TESTS
+echo "" | ../node_modules/.bin/parallel -j "${INTEGRATION_CONCURRENCY:=0}" sh scripts/run_test.sh  $testOutputDir $testTimeout ::: $TESTS
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply example/scripts/tests-parallel.sh:18]: Timeout *does* time out.  It *doesn't* fail, unlike gnu timeout.      I can demo if you'd like.

File: example/scripts/run_test.sh (hunk 1 above)
Comment (Jkovarik) [reply example/scripts/run_test.sh:18]: 😅  so take a look at the exit codes for timeout https://www.gnu.org/software/coreutils/manual/html_node/timeout-invocation.html#timeout-invocation -- the intent here was to log the failure, except I missed this was moving it to passed.   Remedied: 42bd2e4926

Comment (npauzenga) [reply example/scripts/run_test.sh:18]: Ah ok. That makes more sense 😄

**Review by npauzenga — APPROVED** [review]
👍

**Comment by Jkovarik** [comment]
Merging with manual override of CI due to nature of PI, approved by @etcart / @npauzenga  during review meeting.

**Pull request merged**